cmd/insights/contributors.go

```diff
@@ -49,84 +52,132 @@ func NewContributorsCommand() *cobra.Command {
+		insights     = make(contributorsInsightsSlice, 0, len(repositories))
+		allErrors    error
 	)
-	for url := range repositories {
-		repoURL := url
-		waitGroup.Add(1)
-		go func() {
-			defer waitGroup.Done()
-			repoContributorsInsights, err := findAllRepositoryContributorsInsights(context.TODO(), opts, repoURL)
+	go func() {
+		for url := range repositories {
+			repoURL := url
+			waitGroup.Add(1)
+			go func() {
+				defer waitGroup.Done()
+				allData, err := findAllContributorsInsights(ctx, opts, repoURL)
+				if err != nil {
+					errorChan <- err
+					return
+				}
+				if allData == nil {
+					return
+				}
+				insightsChan <- *allData
+			}()
+		}
+		waitGroup.Wait()
+		close(doneChan)
+	}()
+	for {
+		select {
+		case err = <-errorChan:
+			allErrors = errors.Join(allErrors, err)
+		case data := <-insightsChan:
+			insights = append(insights, data)
+		case <-doneChan:
+			if allErrors != nil {
+				return allErrors
+			}
+			output, err := insights.BuildOutput(opts.Output)
 			if err != nil {
-				errorChan <- err
-				return
+				return err
 			}
-			repoContributorsInsights.RenderTable()
-		}()
-	}
-	waitGroup.Wait()
-	close(errorChan)
-	var allErrors error
-	for err = range errorChan {
-		allErrors = errors.Join(allErrors, err)
+			fmt.Print(output)
+			return nil
+		}
 	}
-	return allErrors
 }
 
-type repositoryContributorsInsights struct {
-	RepoID  int32
-	RepoURL string
-	New     []string
-	Recent  []string
-	Alumni  []string
-	Repeat  []string
+type contributorsInsights struct {
+	RepoURL string   `json:"repo_url" yaml:"repo_url"`
+	RepoID  int      `json:"repo_id" yaml:"repo_id"`
+	New     []string `json:"new" yaml:"new"`
+	Recent  []string `json:"recent" yaml:"recent"`
+	Alumni  []string `json:"alumni" yaml:"alumni"`
+	Repeat  []string `json:"repeat" yaml:"repeat"`
 }
 
-func (rci *repositoryContributorsInsights) RenderTable() {
-	if rci == nil {
-		return
-	}
-	rows := []table.Row{
-		{"New contributors", strconv.Itoa(len(rci.New))},
-		{"Recent contributors", strconv.Itoa(len(rci.Recent))},
-		{"Alumni contributors", strconv.Itoa(len(rci.Alumni))},
-		{"Repeat contributors", strconv.Itoa(len(rci.Repeat))},
+type contributorsInsightsSlice []contributorsInsights
+
+func (cis contributorsInsightsSlice) BuildOutput(format string) (string, error) {
+	switch format {
+	case constants.OutputTable:
+		return cis.OutputTable()
+	case constants.OutputJSON:
+		return utils.OutputJSON(cis)
+	case constants.OutputYAML:
+		return utils.OutputYAML(cis)
+	default:
+		return "", fmt.Errorf("unknown output format %s", format)
 	}
-	columns := []table.Column{
-		{
-			Title: "Repository URL",
-			Width: 20,
-		},
-		{
-			Title: rci.RepoURL,
-			Width: len(rci.RepoURL),
-		},
+}
+
+func (cis contributorsInsightsSlice) OutputTable() (string, error) {
+	tables := make([]string, 0, len(cis))
+	for i := range cis {
+		rows := []bubblesTable.Row{
+			{
+				"Repository ID",
+				strconv.Itoa(cis[i].RepoID),
+			},
```

Comment:
We probably don't want/need to surface the internal ID of the repository since that's primarily an implementation detail internal to OpenSauced itself.